File: src/extras-kcoreaddons/libkcoreaddons.cpp
#include <KCoreAddons>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kcoreaddons.h>
#include "libkcoreaddons.h"
#include "libkcoreaddons.hxx"

libqt_string KCoreAddons_VersionString() {
    QString _ret = KCoreAddons::versionString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

unsigned int KCoreAddons_Version() {
    return KCoreAddons::version();
}

File: src/extras-kcoreaddons/libkcoreaddons.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api-staging.kde.org/kcoreaddons.html
pub const kcoreaddons = struct {
    /// [Qt documentation](https://api-staging.kde.org/kcoreaddons.html#versionString)
    ///
    /// ``` allocator: std.mem.Allocator ```
    pub fn VersionString(allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KCoreAddons_VersionString();
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kcoreaddons.VersionString: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api-staging.kde.org/kcoreaddons.html#version)
    ///
    ///
    pub fn Version() u32 {
        return qtc.KCoreAddons_Version();
    }
};
